### ISSUE TITLE
Weird import error when running from venv

### ISSUE THREAD

**Issue by gterzian** [massalabs/massa-network-simulator#37]
When running `python3 main.py` from a venv, I get the below error:

```
Traceback (most recent call last):
  File "main.py", line 9, in <module>
    import cipher
  File "/Users/Gregory/Projects/massa-network-simulator/cipher.py", line 7, in <module>
    from cryptography.hazmat.primitives.ciphers.aead import AESGCM
  File "/Users/Gregory/Projects/massa-network-simulator/lib/python3.8/site-packages/cryptography/hazmat/primitives/ciphers/aead.py", line 10, in <module>
    from cryptography.hazmat.backends.openssl import aead
  File "/Users/Gregory/Projects/massa-network-simulator/lib/python3.8/site-packages/cryptography/hazmat/backends/openssl/__init__.py", line 6, in <module>
    from cryptography.hazmat.backends.openssl.backend import backend
  File "/Users/Gregory/Projects/massa-network-simulator/lib/python3.8/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 63, in <module>
    from cryptography.hazmat.bindings.openssl import binding
  File "/Users/Gregory/Projects/massa-network-simulator/lib/python3.8/site-packages/cryptography/hazmat/bindings/openssl/binding.py", line 14, in <module>
    from cryptography.hazmat.bindings._openssl import ffi, lib
ImportError: dlopen(/Users/Gregory/Projects/massa-network-simulator/lib/python3.8/site-packages/_cffi_backend.cpython-38-darwin.so, 0x0002): symbol not found in flat namespace (_ffi_prep_closure)
```

**Comment by AurelienFT** [comment]
Try to update your python

**Comment by damip** [comment]
Can you try again with the latest corrections ?

**Comment by qdrn** [comment]
Maybe we could include a pipenv to help reproducibility ?

**Comment by gterzian** [comment]
I'm still having this issue, which is further referenced at https://github.com/pyca/cryptography/issues/5742

**Comment by gterzian** [comment]
@AurelienFT Ok, thanks for the info. I propose we then switch to the package that works in all environments(it fixes the issue btw), see https://github.com/massalabs/massa-network-simulator/pull/39

**Comment by AurelienFT** [comment]
Thanks for the PR ! :)

**Comment by gterzian** [comment]
FYI I ran into this issue again when using the latest master which switches back to using `cryptography`, and this time I found the actual fix. 

The problem was that I bootstrapped my new Apple Sillicon Mac with a backup from my old Intel Mac, so the Homebrew linking stuff to Openssl was not working properly. 

I just had to uninstall and re-install Homebrew. 

Also switched to using [Pyenv](https://github.com/pyenv/pyenv) to manage the Python installations. 

I'll add a note to the setup part of the `README`...
